### PR TITLE
Add a trainer & demo notebook for the speech-only pretraining task

### PR DESCRIPTION
**Pull request by lucasnewman**
I cribbed a bunch of this from `SemanticTransformerTrainer` in audiolm-pytorch and added a notebook to demonstrate it works. The loss converges at a 60% mask rate as used in the paper on a subset of LibriTTS. I'm happy to make changes, just let me know!

### CODE REVIEW

**Comment by lucidrains** [comment]
Great job Lucas! I'll take a look later this week; I'm about to dive back into the TTS field in August and finish a bunch of repos.

**Comment by lucasnewman** [comment]
> @lucasnewman are you doing this for work? for a company in SF perhaps?

Yep, it's part of an exploration I'm doing for work and also just advancing my understanding of the SOTA along the way.

**Comment by lucidrains** [comment]
@lucasnewman cool, maybe my dog and I will run into you :laughing:  we live in the mission

**Comment by lucidrains** [comment]
@lucasnewman which company do you work for?

**Comment by lucasnewman** [comment]
> @lucasnewman which company do you work for? just curious if it is yet another TTS company (been contacted by like 3 so far)

Ha, not at all, I work for [Future](https://www.future.co) (you will probably be confused 😅). I'm over in Noe so not too far away!

**Comment by lucasnewman** [comment]
> @lucasnewman haha yea i am confused 😆 you automating the personal trainer with some deep fake? nice! Vaswani lives in Noe Valley haha (great neighborhood)

It's not really to replace the humans, but more around personalizing the other audio aspects of what we do — I find the vast majority of deep fakes still fully in the uncanny valley. I see Vaswani at La Lucha on Sanchez all the time, although I don't know him. Small world for sure!

**Comment by lucidrains** [comment]
@lucasnewman yes, SF is indeed a very small world filled with some giants in the industry

ok! i will take you up on your offer later this month! expect TTS to be in a very good place (open source wise) by end of month or mid September

**Comment by lucasnewman** [comment]
I'm going to close this one in favor of https://github.com/lucidrains/spear-tts-pytorch/pull/4, since that has all the changes here and more support for backtranslation. 🙏
